### ISSUE TITLE
Implement Different learning algorithm from scratch with visualization

### ISSUE THREAD

**Issue by iamarpan** [gimseng/99-ML-Learning-Projects#84]
One of the most important component of neural networks is the learning algorithms it uses.
Since for most of the beginners it is like a black box.
I am planning to contribute code for these learning algorithms. These will include:
1) Gradient Descent
2) Momentum based Gradient Descent
3) Nesterov Accelarated Gradient Descent
4) AdaGrad
5) RMSProp
6) Adam
It will be a .ipynb file with complete implementation from scratch also with proper visualization as well as documentation, so that beginners can get a better understanding of these concepts.

**Comment by gimseng** [comment]
@iamarpan Good idea! Please go ahead, but do read the contributing guidelines and look at one or two of the latest project folders. Many thanks !

**Comment by iamarpan** [comment]
@gimseng Sorry was a bit occupied with office work. Can I pick this one now?
